sdk/identity/azure-identity/azure/identity/_credentials/default.py

```diff
@@ -102,6 +105,9 @@ def __init__(self, **kwargs):
         managed_identity_client_id = kwargs.pop(
             "managed_identity_client_id", os.environ.get(EnvironmentVariables.AZURE_CLIENT_ID)
         )
+        interactive_browser_client_id = kwargs.pop("interactive_browser_client_id", None)
+        if not isinstance(interactive_browser_client_id, six.string_types):
+            raise ValueError('"interactive_browser_client_id" must be a string')
```

Comment:
But the variable defaults to `None`, causing `DefaultAzureCredential()` to raise this error.

Reply:
Hah, Good catch!.

Reply:
How about:
```py

browser_args = dict(kwargs, tenant_id=interactive_browser_tenant_id)
interactive_browser_client_id = kwargs.pop("interactive_browser_client_id", None)
if interactive_browser_client_id:
    browser_args["client_id"] = interactive_browser_client_id
...
if not exclude_interactive_browser_credential:
    credentials.append(InteractiveBrowserCredential(**browser_args))
```

Reply:
We want to pass in the client_id after validating client_id is a string, right?

Are you proposing we doing it in ctor of InteractiveBrowserCredential?

Reply:
We want to pass client_id if and only if the caller gave us a value for interactive_browser_client_id. My code snippet is one way to achieve that. Only a string will work for client_id but in general we don't validate types and there's no special need to do so here.

Reply:
I think we are on same page and I guess it is already addressed?

(please refresh the page, I moved the validation into 

```py
if not exclude_interactive_browser_credential:
    if interactive_browser_client_id:
```
)

Reply:
Yes, that's another way to do it. No need for the `isinstance` though.